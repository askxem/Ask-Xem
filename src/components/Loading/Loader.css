.loader {
  width: 10px;
  height: 10px;
  margin: 40px;
  border-radius: 50%;
  background: #e57c49;
}

.container {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: center;
}
