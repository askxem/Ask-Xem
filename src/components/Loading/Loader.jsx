import React from 'react';
import { motion } from 'framer-motion';
import styles from './Loader.css'

const loaderVariants ={
    animationOne: {
        x: [-30, 30],
        y: [0, -30],
        transition: {
            x: {
                yoyo: Infinity,
                duration: 0.5
            },
            y: {
                yoyo: Infinity,
                duration: 0.25
            },
        }
    }
}

export default function Loader() {

    return(
        <div className={styles.container}>
            <motion.div className={styles.loader} variants={loaderVariants} animate='animationOne' aria-label='Loading.'></motion.div>
        </div>
    );
}
