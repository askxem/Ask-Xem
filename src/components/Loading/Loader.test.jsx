import { render } from '@testing-library/react';
import Loader from './Loader.jsx';

it('renders loader component', () => {
    const { container } = render(
        <Loader />
    )

    expect(container).toMatchSnapshot();
})