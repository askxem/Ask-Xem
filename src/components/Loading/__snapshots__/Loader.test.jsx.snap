// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders loader component 1`] = `
<div>
  <div
    class="container"
  >
    <div
      aria-label="Loading."
      class="loader"
    />
  </div>
</div>
`;
